go/runtime/registry/registry.go

```diff
@@ -599,12 +584,12 @@ func New(ctx context.Context, dataDir string, consensus consensus.Backend, ias i
 		runtimes:  make(map[common.Namespace]*runtime),
 	}
 
-	switch cfg.Mode {
-	case RuntimeModeNone:
+	switch config.GlobalConfig.Mode {
+	case config.ModeValidator, config.ModeSeed:
 		r.logger.Info("runtime support is disabled")
 	default:
 		r.logger.Info("runtime support is enabled",
-			"mode", cfg.Mode,
+			"mode", config.GlobalConfig.Mode,
```

Comment:
Given that this is now global, it should probably be logged during node startup and removed from here (this entire `switch`).